rust/src/sysroot_upgrade.rs

```diff
@@ -0,0 +1,28 @@
+//! Rust portion of `rpmostree-sysroot-upgrader.cxx`.
+
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::cxxrsutil::*;
+use anyhow::{Context, Result};
+use std::convert::TryInto;
+use std::pin::Pin;
+
+/// Import ostree commit in container image using ostree-rs-ext's API.
+pub fn import_container(
+    mut sysroot: Pin<&mut crate::FFIOstreeSysroot>,
+    imgref: String,
+) -> Result<String> {
+    let sysroot = &sysroot.gobj_wrap();
+    let repo = &sysroot.get_repo(gio::NONE_CANCELLABLE)?;
```

Comment:
Seeing this reminds me actually...we should make the container import cancellable.  Don't try to fix this right now, but let's add something like `// TODO: take a GCancellable and monitor it, and drop the import task (which is how async cancellation works in Rust)`?

Right now if one does e.g. `rpm-ostree upgrade` and we pull a new container image, then they type `Ctrl-C` to cancel, that triggers the `GCancellable`.  We need to bridge the GCancellable into the Rust async world.